src/components/media-player/ha-media-player-browse.ts

```diff
@@ -564,6 +573,48 @@ export class HaMediaPlayerBrowse extends LitElement {
     this._resizeObserver.observe(this);
   }
 
+  /**
+   * Load thumbnails for images on demand as they become visible.
+   */
+  private async _attachInteractionObserver(): Promise<void> {
+    if (!this._thumbnails) {
+      return;
+    }
+    if (!this._interactionObserver) {
+      this._interactionObserver = new IntersectionObserver(
+        async (entries, observer) => {
+          await Promise.all(
+            entries.map(async (entry) => {
+              if (!entry.isIntersecting) {
+                return;
+              }
+              const thumbnailCard = entry.target as HaCard;
+              let thumbnailUrl = thumbnailCard.dataset.src;
+              if (!thumbnailUrl) {
+                return;
+              }
+              if (thumbnailUrl.startsWith("/")) {
+                // Thumbnails served by local API require authentication
+                thumbnailUrl = await getSignedThumbnailPath(
+                  this.hass,
+                  thumbnailUrl
+                );
+              }
+              thumbnailCard.style =
```

Comment:
Does it make sense to have a separate LitElement that is a lazy loading image? Then the media player can change state on it, and it re-renders.

Reply:
nah this is fine. You just need to use `thumbnailCard.setAttribute("style", "value")` if you want to set the attribute, or use `thumbnailCard.style.backroundImage = \`url(${thumbnailUrl})\`` if you want to use properties (I would use properties)